etc/taskcluster/decisionlib.py

```diff
@@ -57,6 +57,19 @@ def __init__(self):
         self.git_ref = os.environ.get("GIT_REF")
         self.git_sha = os.environ.get("GIT_SHA")
 
+        # If the head commit is a merge, we want a generate a unique task id which incorporates
+        # the merge parents rather that the actual sha of the merge commit. This ensures that tasks
+        # can be reused if the tree is in an identical state. Otherwise, if the head commit is
+        # not a merge, we can rely on the head commit sha for that purpose.
+        last_merge_output = subprocess.check_output(["git", "log", "--merges", "--format=%H", "-1"])
+        print(last_merge_output)
+        if last_merge_output.decode("utf8").strip() == self.git_sha:
+            merge_parents = subprocess.check_output(["git", "show", "--format=%P", "HEAD"])
+            print(merge_parents)
+            self.task_id = '-'.join(merge_parents.decode("utf8").strip().split(' '))
```

Comment:
Maybe we could run `git show` first, and look at the length of the result of `.split(' ')` here instead of running `git log --merges`?